pootle/apps/pootle_store/forms.py

```diff
@@ -280,11 +284,16 @@ def clean_target_f(self):
 
             return value
 
-        def clean_state(self):
+        def _clean_state(self):
+            # Handle clean_state here because it depends on clean_suggestion.
             old_state = self.instance.state  # Integer
             is_fuzzy = self.cleaned_data['state']  # Boolean
             new_target = self.cleaned_data['target_f']
 
+            # If suggestion is provided set `old_state` should be `TRANSLATED`.
+            if self.cleaned_data['suggestion']:
+                old_state = TRANSLATED
+
             if (self.request is not None and
                 not check_permission('administrate', self.request) and
                 is_fuzzy):
```

Comment:
...but whatever you do - please dont put multi-line conditions into conditional blocks, as the indentation is very unclear, (ignore if you really must)